oeps/oep-0051-bp-conventional-commits.rst

```diff
@@ -0,0 +1,184 @@
+
+Scope
+-----
+
+The `Conventional Commits`_ spec includes a parenthesized scope after the type.  Open edX repos are large and varied, making standardization of scopes difficult.  We won't use scopes for now.  This could change in the future.
+
+Subject
+-------
+
+Commit message subjects should be short enough to fit on one line.  We aren't putting a hard limit on character length, but 70 characters is a good time to turn your attention to the body of the commit message.  Put more information in the body of the commit message to fully explain your change.  Jira or GitHub issue numbers can be included at the end of the subject.
+
+Body
+----
+
+The subject of the commit is rarely enough information to fully understand the commit.  The body can contain as much information as you like.  Be generous.  Take a moment to think about what you would want to know if someone else had authored this commit.
+
+The more information you can put in the body of the commit message, the better. It's not unreasonable to have two paragraphs of explanation in the body.  Especially important are the reasons for the change, or other factors that won't be apparent from the code itself.
+
+Breaking changes especially should have detailed information about the implications and alternatives.
+
+
+Footer
+------
+
+We are not requiring any formalized footers in the commit message, though the use of  a ``BREAKING CHANGE:`` footer is encouraged as a way of providing more details about breaking changes.
+
+
+Discussion
+==========
+
+Conventional Commits asks us to categorize changes into a small number of categories.  There will naturally be cases where it isn't obvious which commit type to use.
+
+Choosing the commit type to use for a commit is important, but it's only the first step.  If you have doubts about which commit type to use, choose the highest-priority type that could apply.  Then write a detailed body explaining the full complexity of your change.
+
+**feat vs fix**: some user-visible changes to features could be classified as "feat" or "fix".  Choose "feat" if the change adds to the set of features.  Choose "fix" if the change affects how a feature behaves.  Yes, this is still subjective.  
+
+**breaking changes to features**: changing how a feature works is not a breaking change.  For example, users are sent to a new experience instead of the old experience. This is not a breaking change.  It should get a "feat" label, but not a "feat!" label.
+
+**DEPR**: removing deprecated code likely is a breaking change.
+
+**pinned dependencies**: updating the version of a pinned dependency seems like just a chore, but consider the repo consumer's perspective.  If an updated dependency adds a feature, then the one-line commit to update the pinned version should be marked "feat".
```

Comment:
This is really important, because I see a lot of feature changes arrive in edx-platform under the commit message "Updating Python Requirements" 

But I fear it's going to be hard for the committer of a dependency update to manage. 

Consider this commit https://github.com/edx/edx-platform/pull/26589/commits/7c9513d2aed42d1689ebb3eea1c843d8f626dbe0 -- I believe the committer was intending to update astroid but they also picked up an update to the edx-sga xBlock (a bug fix in this case).  

Should the committer be responsible for tracking down the nature of each of the dependency updates? Or do they have a way to scope their commit so it doesn't go beyond the chore they intended?

Reply:
I'd imagine that the "chore" of updating python dependencies could pull in newer versions of multiple repos.  It's likely that one or more of the sub-repos could have `feat`, `revert`, `docs`, etc changes in them. I don't believe there's information-value in having dependency-upgrade PRs muddy the semantics of the change by (for example) marking with the "maximum" of the commit-types. As a consumer, you lose information that these changes were brought in through upgrades.

I wonder if better tooling to visualize/summarize the commits taken from the sub-repos could help here insetad. CCing @jmbowman and @awais786 since their team is likely impacted and can provide ideas and input.

Reply:
This I think is going to be a real challenge.  The entire motivation for conventional commits is to make it possible to easily understand the scope of a change from the commit message.  Commits like "chore: automated requirements update" will be a huge gap where significant changes can appear without indication.  I'm not sure what to do about this.

Reply:
I defer to @jmbowman on this.  I'm good with what is feasible for Arbi-BOM's continual-upgrade efforts.